### PR TITLE
#219 Collapsing of subQuery properties

### PR DESCRIPTION
**Pull request by amaleszewski**
Hello, I've created a PR for collapsing a subQuery properties as I've mentioned in #219 issue.

### CODE REVIEW

File: config-example.yml [hunk 1]
```diff
@@ -268,6 +268,10 @@ extensions:
   # thing to have on, but it is possible to turn it off.
   # Default: true
   graphqlScalarExamples: true
+  # It defines from which subQuery depth, the subqueries should be collapsed as a Fragments in preview
+  # instead of showing all properties.
+  # Default: 1
+  collapseSubQueryDepth: 3
```
Comment (newhouse) [inline config-example.yml:274]: I think this should be named something else like `typeExpansionDepth`.

And I think it should be placed under `introspection` options, and not the `extensions` area.

File: src/index.js [hunk 2]
```diff
@@ -114,6 +114,7 @@ const introspectionOptionDefaults = Object.freeze({
 
 const extensionsOptionDefaults = Object.freeze({
   graphqlScalarExamples: true,
+  collapseSubQueryDepth: 1
```
Comment (newhouse) [inline src/index.js:117]: This will need to be moved to the `introspectionOptionDefaults`, please.

**Review by newhouse — CHANGES_REQUESTED** [review]
@amaleszewski thank you so much for this PR!

I love that you've submitted this rather than just keep it to yourself as part of a fork, and I love this option.

I do just have a few requests around where the option belongs, but other than that I'm willing to accept it after those changes.

I'd love to see a test for this if possible, but there's no pattern or existing module for this area of the code so far, so that's probably a bit much to ask of you...and I'll have to add it myself at some point.

Thanks again!

**Comment by amaleszewski** [comment]
Thanks for response, I will try to fix that in the next week according to the remarks you added, because I don't have time in this week.

I'm also not sure about adding new tests, because it could be time-consuming for me but I will try :)

**Comment by newhouse** [comment]
> Thanks for response, I will try to fix that in the next week according to the remarks you added, because I don't have time in this week.
> 
> I'm also not sure about adding new tests, because it could be time-consuming for me but I will try :)

In the interest of time, I have taken your work and just brought it across the finish line myself in another PR:
https://github.com/anvilco/spectaql/pull/302

I'll close this PR now as this change has been merged and is out in `1.1.0`.

Thanks for your contribution, @amaleszewski !!!

**Pull request closed without merging**